unit-test/background/reference-tests/broken-site-reporting-tests.js

```diff
@@ -64,7 +64,7 @@ for (const setName of Object.keys(testSets)) {
 
                 if (test.expectReportURLParams) {
                     test.expectReportURLParams.forEach(param => {
-                        expect(requestURLString).toContain(`${param.name}=${param.value}`)
+                        expect(requestURLString).toMatch(new RegExp(`[?&]${param.name}=${param.value}[&$]`))
```

Comment:
tests were not catching the surrogates/blockedTrackers issue because I used too simple method of matching params, regex matching fixes it

Reply:
`param.value` is already URLencoded here right? I wonder how safe it is to just drop this into a Regex unsanitized? Are there any Regex special characters that could come through URL encoding? Are we OK allowing the reference test repo to run any regex here if it wants to?

Reply:
Yep, param value is already encoded in tests. I don't think there is any security risk here (we control all inputs, this is test code), but I agree that it can happen that we will write a test that has some special chars and end up with unexpected results. I'll fix 👍